docs/cruisecontroloperation.md

```diff
@@ -0,0 +1,320 @@
+
+```yaml
+status:
+...
+  brokersState:
+    "3":
+     ...
+      gracefulActionState:
+        cruiseControlOperationReference:
+          name: kafka-addbroker-mhh72
+        cruiseControlState: GracefulUpscaleCompletedWithError
+    ...
+```
+
+The kafka-addbroker-mhh72 `CruiseControlOperation` from the above example:
+
+```yaml
+kind: CruiseControlOperation
+metadata:
+...
+  name: kafka-addbroker-mhh72
+...
+spec:
+...
+status:
+  currentTask:
+    finished: "2022-11-18T09:31:40Z"
+    httpRequest: http://kafka-cruisecontrol-svc.kafka.svc.cluster.local:8090/kafkacruisecontrol/add_broker?allow_capacity_estimation=true&brokerid=3&data_from=VALID_WINDOWS&dryrun=false&exclude_recently_demoted_brokers=true&exclude_recently_removed_brokers=true&json=true&use_ready_default_goals=true
+    httpResponseCode: 200
+    id: 222e30f0-1e7a-4c87-901c-bed2854d69b7
+    operation: add_broker
+    parameters:
+      brokerid: "3"
+      exclude_recently_demoted_brokers: "true"
+      exclude_recently_removed_brokers: "true"
+    started: "2022-11-18T09:30:48Z"
+    state: Completed
+    summary:
+      Data to move: "0"
+      Intra broker data to move: "0"
+      Number of intra broker replica movements: "0"
+      Number of leader movements: "0"
+      Number of replica movements: "36"
+      Provision recommendation: '[ReplicaDistributionGoal] Remove at least 4 brokers.'
+      Recent windows: "1" 
+  errorPolicy: retry
+  retryCount: 0
+```
+
+The `status.currentTask` describes the Cruise Control task. The `httpRequest` field contains the whole POST HTTP request which has been executed. The `id` is the Cruise Control task identifier number. The `state` shows the progress of the add_broker request. The `summary` is the Cruise Control's optimization proposal. It shows the scope of the changes that Cruise Control will be applied through the operation. When a task execution completed with an error and there are re-executions the `retryCount` shows the number of the retries. In this case the history of the failed tasks are visible with the error message in the `status.failedTask` field list.  
+You can also find information from the fields [here](https://github.com/banzaicloud/koperator/blob/master/api/v1alpha1/cruisecontroloperation_types.go)
+
+## How can we have control over the created CruiseControlOperation?
+
+- The task execution can be stopped gracefully when the `CruiseControlOperation` is deleted. In this case the corresponding `cruiseControlState` or the `cruiseControlVolumeState` will get succeeded state.
```

Comment:
```suggestion
- The task execution can be stopped gracefully when the `CruiseControlOperation` is deleted. In this case the corresponding `cruiseControlState` or the `cruiseControlVolumeState` will become `succeeded`.
```

Reply:
I think we should not highlight the `succeeded` word.

Reply:
Yeah, I guess you may right, I highlighted it because I was guessing that was what the state was called. If that's not the case then my sentence doesn't even make sense here anyway. :D